frontend/src/stores/useBiolink.js

```diff
@@ -0,0 +1,124 @@
+import _ from 'lodash';
+import strings from '~/utils/strings';
+
+const baseClass = 'biolink:BiologicalEntity';
+
+export default function useBiolink() {
+  const [biolink, setBiolink] = useState(null);
+  const [concepts, setConcepts] = useState([]);
+  const [descendancies, setDescendancies] = useState({});
+  const [predicates, setPredicates] = useState([]);
+
+  function getEdgePredicates() {
+    const newPredicates = Object.entries(biolink.slots).map(([identifier, predicate]) => ({
+      predicate: strings.edgeFromBiolink(identifier),
+      domain: strings.nodeFromBiolink(predicate.domain),
+      range: strings.nodeFromBiolink(predicate.range),
+    }));
+    return newPredicates;
+  }
+
+  /**
+   * Given a biolink class, build a list of parent classes
+   * Ex: Given 'gene' as an input this function will return:
+   *
+   * [ "gene", "gene_or_gene_product", "macromolecular_machine",
+   *   "genomic_entity", "molecular_entity", "biological_entity",
+   *   "named_thing" ]
+   *
+   * @param {string} className - a biolink class
+   * @param {object} biolinkClasses - object of all biolink classes
+   * @returns {string[]} list of related biolink classes
+   */
+  function getDescendants(childClass, classes) {
+    let currentClass = childClass;
+    const descendantList = [currentClass];
+    // Repeat until we hit the top of the classes
+    while (
+      classes[currentClass] &&
+      classes[currentClass].is_a
+    ) {
+      // reassign current type to parent type
+      currentClass = classes[currentClass].is_a;
+      descendantList.push(currentClass);
+    }
+    return descendantList;
+  }
+
+  /**
+   * Get a list of descendants for all biolink classes
+   * @param {object} biolinkClasses - object of all biolink classes
+   * @returns {{}} Object with classes as keys and descendant lists as values
+   */
+  function getAllDescendancies(biolinkClasses) {
+    const newHierarchies = Object.keys(biolinkClasses).reduce((obj, item) => {
+      let hierarchy = getDescendants(item, biolinkClasses);
+      hierarchy = hierarchy.map((h) => strings.nodeFromBiolink(h));
+      obj[strings.nodeFromBiolink(item)] = hierarchy;
+      return obj;
+    }, {});
+    return newHierarchies;
+  }
+
+  /**
+   * Filter out concepts that are not derived classes of base class
+   * @param {object} allDescendancies - object of all descendant lists
+   * @returns {array} list of valid concepts
+   */
+  function getValidConcepts(allDescendancies) {
+    const newConcepts = Object.keys(allDescendancies).filter((biolinkClass) => allDescendancies[biolinkClass].includes(baseClass));
+    return newConcepts;
+  }
+
+  /**
+   * Get all direct descendants of a target class, recursively
+   * @param {object} classes - all biolink classes
+   * @param {*} targetClass - class we want to find the descendants of
+   * @returns {[]} list of descendants
+   */
+  function getDirectDescendantsRecursive(classes, targetClass) {
```

Comment:
I'd name this according to what it does, rather than how it does it. What is the actual difference between what `getDirectDescendantsRecursive()` returns and what `getAllDescendants()` returns? The format of the class names?

Reply:
We get hierarchies two ways. We get all ascendants for all classes below `named_thing`, but for named_thing, we need to get all descendants. Updated in f22b4412f0713e88b7703704c15d0201a60916d1